### PR TITLE
change redis commands option type to pointer

### PR DESCRIPTION
**Pull request by cep-ter**

### CODE REVIEW

File: commander/commander.go [hunk 1]
```diff
@@ -341,24 +355,27 @@ func (c *Commander) XClaim(result interface{}, streamName, groupName, consumerNa
...
-		cmd = cmd.Add("force")
-	}
-	if options.Justid {
-		cmd = cmd.Add("justid")
+	if options != nil {
+		if options.Idle != 0 {
+			cmd = cmd.Add("IDLE")
+			cmd = cmd.Add(options.Idle.Milliseconds())
+		}
+		if options.Time.Unix() > 0 {
```
Comment (bbmokhtari) [inline commander/commander.go:363]: Please use the standard Redis interface integer instead of time types.

File: commander/commander.go [hunk 2]
```diff
@@ -341,24 +355,27 @@ func (c *Commander) XClaim(result interface{}, streamName, groupName, consumerNa
...
+			cmd = cmd.Add("TIME")
+			cmd = cmd.Add(options.Time.Unix())
+		}
+		if options.RetryCount != 0 {
```
Comment (bbmokhtari) [inline commander/commander.go:367]: Please check the inconsistencies for types, specifically `int`. and also check the inconsistencies for operators, specially `>` and `!=`.

**Review by bbmokhtari — CHANGES_REQUESTED** [review]
Please check the comment and elaborate.

**Review by bbmokhtari — APPROVED** [review]

**Review by rezakamalifard — CHANGES_REQUESTED** [review]

File: commander/commander_test.go [hunk 3]
```diff
@@ -235,7 +235,7 @@ func TestXadd(t *testing.T) {
 	cmd := New(conn)
 	var xaddResult string
 	errCmd := cmd.
-		XAdd(&xaddResult, streamNmae, "*", map[string]string{key: value}, XAddOption{}).
+		XAdd(&xaddResult, streamNmae, "*", map[string]string{key: value}, nil).
```
[the remark this answers is not in the document]
Comment (cep-ter) [reply commander/commander_test.go:238]: The main purpose of this pr is to replace empty options with nil.
If you think this is not clear maybe we shouldn't change option type to pointer.

File: commander/commander_bdd_test.go [hunk 4]
```diff
@@ -1463,7 +1463,7 @@ var _ = Describe("Commander", func() {
 
 			errCmd := commander.
 				Select(&selectResult, 0).
-				Set(&setResult, key, 9, SetOption{}).
```
[the remark this answers is not in the document]
Comment (cep-ter) [reply commander/commander_bdd_test.go:1466]: I try to make it safer in run-time by restrict user to predefined option structs rather than variadic args

**Pull request closed without merging**